tests/std/tests/P0896R4_ranges_subrange/test.cpp

```diff
@@ -0,0 +1,1223 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+// Covers ranges::view_interface and ranges::subrange
+
+#include <cassert>
+#include <concepts>
+#include <forward_list>
+#include <list>
+#include <ranges>
+#include <string_view>
+#include <tuple>
+#include <type_traits>
+#include <utility>
+#include <vector>
+//
+#include "range_algorithm_support.hpp"
+
+#define ASSERT(...) assert((__VA_ARGS__))
+
+using std::output_iterator_tag, std::input_iterator_tag, std::forward_iterator_tag, std::bidirectional_iterator_tag,
+    std::random_access_iterator_tag, std::contiguous_iterator_tag;
+
+int main() {} // COMPILE-ONLY
+
+struct empty {};
+
+namespace test_view_interface {
+    template <class T>
+    concept CanViewInterface = requires {
+        typename ranges::view_interface<T>;
+    };
+
+    template <class T>
+    constexpr bool test_template_id() {
+        // view_interface<T> is a valid template-id only if T is a cv-unqualified class type
+        STATIC_ASSERT(std::is_same_v<T, std::remove_cvref_t<T>>);
+        STATIC_ASSERT(CanViewInterface<T> == std::is_class_v<T>);
+        if constexpr (!std::is_function_v<T>) {
+            STATIC_ASSERT(!CanViewInterface<T const>);
+            STATIC_ASSERT(!CanViewInterface<T volatile>);
+            STATIC_ASSERT(!CanViewInterface<T const volatile>);
+        }
+        if constexpr (!std::is_void_v<T>) {

```

Comment:
It's test code, but consider adding a newline to separate non-chained control flow.